### PR TITLE
zprint: init at 1.1.2

### PR DESCRIPTION
**Pull request by stelcodes**
###### Motivation for this change
Adding zprint to nixpkgs, a clojure pretty printer and source code formatter.

###### Things done
Added myself to the maintainer list, added derivation for zprint

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stelcodes** [comment]
Hey! This is my first attempt at not only NixOS packaging, but linux packaging in general. I based this derivation on the babashka derivation, since they are both built with graalvm-ce. One thing that was confusing was that babashka uses graalvm11-ce.meta.platforms which evaluates to [ "x86_64-linux" "x86_64-darwin" ] but on search.nixos.org babashka platforms appear as only "x86_64-linux". How does that work? Both packages can only build on x86_64-linux I believe because GraalVM requires a license to build a native-image on Darwin.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9481,6 +9481,12 @@
     githubId = 1699155;
     name = "Steve Elliott";
   };
+  stelcodes = {
+    email = "stel@stel.codes";
+    github = "stelcodes";
+    githubId = 22163194;
+    name = "Stel Abrego";
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:9489]: This goes into a separate commit: `maintainers: add stelcodes`

File: pkgs/development/tools/zprint/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+with lib;
+stdenv.mkDerivation rec {
+  pname = "zprint";
+  version = "1.1.2";
+
+  src = fetchurl {
+    url =
+      "https://github.com/kkinnear/${pname}/releases/download/${version}/${pname}-filter-${version}";
+    sha256 = "1wh8jyj7alfa6h0cycfwffki83wqb5d5x0p7kvgdkhl7jx7isrwj";
+  };
```
Comment (legendofmiracles) [inline pkgs/development/tools/zprint/default.nix:12]: use fetchFromGitHub instead

File: pkgs/development/tools/zprint/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    -H:EnableURLProtocols=https,http \
+    -H:+ReportExceptionStackTraces \
+    --report-unsupported-elements-at-runtime \
+    --initialize-at-build-time \
+    --no-fallback
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${pname} $out/bin/${pname}
```
Comment (legendofmiracles) [inline pkgs/development/tools/zprint/default.nix:35]: can be replaced with the `install` command

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 123150` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zprint</li>
  </ul>
</details>

File: pkgs/development/tools/zprint/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{ stdenv, lib, fetchurl, graalvm11-ce, glibcLocales }:
+
+with lib;
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/zprint/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/tools/zprint/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    --initialize-at-build-time \
+    --no-fallback
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${pname} $out/bin/${pname}
+  '';
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/zprint/default.nix:38]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by lukegb** [comment]
Thanks for the contribution! Do you mind making the changes above (splitting out the maintainer addition to a separate commit) as well as rebasing to remove the merge commit?

**Pull request merged**